p4-16/spec/P4-16-spec.mdk

```diff
@@ -6885,6 +6908,40 @@ invokes action `a`. Once the program is loaded, these entries are
 installed in the table in the order they are enumerated in the
 program.
 
+##### Entry priorities { #sec-entry-priorities }
+
+If a table has fields where their `match_kind`s are all `exact` or
+`lpm`, there is no reason to assign numeric priorities to its entries.
+If they are all `exact`, duplicate keys are not allowed, and thus
+every lookup key can match at most one entry, so there is no need for
+a tie-breaker.  If there is an `lpm` field, the priority of the entry
+corresponds to the length of the prefix, i.e. if a lookup key matches
+multiple prefixes, the longest prefix is always the winner.
+
+For tables with other `match_kind` values, e.g. at least one `ternary`
+field, in general it is possible to install multiple entries such that
+the same lookup key can match the key of multiple entries installed
+into the table at the same time.  Control plane APIs such as P4Runtime
+API[^P4RuntimeAPI] and TDI[^TDI] require control plane software to
+provide a numeric priority with each entry added to such a table.
+This enables the data plane to determine which of several matching
+entries is the "winner", i.e. the one entry whose action is invoked.
+
+[^TDI]: TDI is the Table Driven Interface.  More information can be
+    found here: https://github.com/p4lang/tdi
+
+Unfortunately there are two commonly used, but different, ways of
+interpreting numeric priority values.
+
+The P4Runtime API requires numeric priorities to be positive integers,
+and define that entries with larger priorities must win over
+entries with smaller priorities.  We will call this convention
+`largest_priority_wins`.
+
+TDI requires numeric priorities to be non-negative integers, and
+define that entries with smaller priorities must win over entries with
+larger priorities.
```

Comment:
How is this convention called?? (the P4Runtime strategy is called 'largest-priority_wins', so I'm curious what is this called)

Reply:
Well largest_priority_wins is just the name I'm using in this document, to give a name to it.  I would call TDI smallest_priority_wins.